src/Core/Services/RequestValidator.cs

```diff
@@ -546,17 +546,17 @@ private static StoredProcedureDefinition TryGetStoredProcedureDefinition(string
         /// </summary>
         /// <param name="first">String representing value associated with $first</param>
         /// <returns>uint > 0 representing $first</returns>
-        public static uint CheckFirstValidity(string first)
+        public static int CheckFirstValidity(string first)
         {
-            if (!uint.TryParse(first, out uint firstAsUint) || firstAsUint == 0)
+            if (!int.TryParse(first, out int firstAsInt) || firstAsInt == 0)
             {
                 throw new DataApiBuilderException(
-                        message: $"Invalid number of items requested, $first must be an integer greater than 0. Actual value: {first}",
+                        message: $"Invalid number of items requested, $first must be -1 or an integer greater than 0. Actual value: {first}",
```

Comment:
`(!int.TryParse(first, out int firstAsInt) || firstAsInt == 0)` seems to allow for values below -1, do we need to change the responsibility of this helper function?

Reply:
Thanks for catching this. Let me fix.